src/pydap/tests/test_client.py

```diff
@@ -1308,3 +1323,50 @@ def test_data_check(var_batch, key_batch, var_name, var_key, expected_shape):
     data = data_check(var, var_key)
 
     assert data.shape == expected_shape
+
+
+def test_consolidate_metadata_non_batch(cache_tmp_dir):
```

Comment:
@jgallagher59701 This is the test that matters.

Both files have the same identical dimensions COADSX and COADSY. The method `consolidate_metadata` creates a special key so that these are downloaded from only 1 granule (separte dap url). And for any additional URL that has the same COADSX dimension and has a nearly identical url, it reuses the cached response for COADSX .


The last assertion proofs that. Even then the dap url is not on the cache, it reads data from the cache.

Reply:
This is a tricky optimization - because it could be 'fooled' by badly  formed data. But it fits with some much of the multi-file data we see that it makes sense. This was used in NcML and the TDS when it did/does aggregations and also our NcML handler in Hyrax, although in those cases the data provider set that up through configuration.

Is this a behavior that can be turned on/off by the user/client?

Reply:
Yes - if a user skips the `consolidate_metadata` step, and simply do:

```python
ds = xr.open_dataset(...)
```
then none of the "optimization" takes place. 

And restarting a session (for example, when restarting a jupyter kernel) also removes all of these optimization

```python
session = CashedSession(cache_name='name_to_db')
```
that will restart the session, pointing to the databaset, but without any of the cache_key optimization. It will be significantly slow still, since xarray will download all the dimensions for each granule (which are not stored originally in the database)

Reply:
OK, that makes sense - it's there if they ask for it, but not there by default. So they must be aware. Good.